templates/explore/repos.tmpl

```diff
@@ -2,7 +2,7 @@
 <div role="main" aria-label="{{.Title}}" class="page-content explore repositories">
 	{{template "explore/navbar" .}}
 	<div class="ui container">
-		{{template "explore/repo_search" .}}
+		{{template "explore/repo_search" dict "ctxData" . "AutoFocus" true}}
```

Comment:
hmm is there no better way to pass args to a template?

Reply:
That's the "correct" way to do it I think 👀

Reply:
It's not ideal and quite fragile.

For example, this PR is still incomplete and there are regression bugs caused by this change: `{{$.SortType}}`

There could be some better approaches to improve the maintainability and stability, eg: pass some parameters explicitly, etc.

Reply:
[(maybe) a better way to set autofocus](https://github.com/go-gitea/gitea/pull/27982/commits/fc841f23f0ce0c2547682ad3780f4e781f407865)

Reply:
Previously I had added a `MergeInto` helper which arguably was a bit better than this as it did not require changes in the subtemplate (it merged additional props into the `.` context), but if I recall correctly, it was removed by @wxiaoguang in a refactor.

Reply:
* https://github.com/go-gitea/gitea/pull/23932

The `MergeInto` is a redundant helper, and the "merge" behavior would cause fragile code. That's why the "merge" behavior is **discouraged**. It's always good to use explicit keys.

<img width="1000" alt="image" src="https://github.com/go-gitea/gitea/assets/2114189/101c30fe-1469-424c-8a1c-72d42e18f88f">

Reply:
Yes, at the expense of diff in the subtemplate. We could avoid this by not using `.` as well.

Reply:
But I see some other solution was found, so lgtm.